show-geodesic-path-between-two-points/src/main/java/com/esri/arcgismaps/sample/showgeodesicpathbetweentwopoints/MainActivity.kt

```diff
@@ -0,0 +1,193 @@
+    private val graphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create and add a map with a navigation night basemap style
+        val map = ArcGISMap(BasemapStyle.ArcGISImageryStandard).apply {
+            initialViewpoint = Viewpoint(Point(34.77, -10.24), 20e7)
+        }
+        // configure mapView assignments
+        mapView.apply {
+            this.map = map
+            // add the graphics overlay to the mapview
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        // add all our marker graphics to the graphics overlay
+        graphicsOverlay.graphics.addAll(
+            listOf(startingLocationMarkerGraphic, endLocationMarkerGraphic, geodesicPathGraphic)
+        )
+
+        lifecycleScope.launch {
+            // check if the map has loaded successfully
+            map.load().onSuccess {
+                // capture and collect when the user taps on the screen
+                mapView.onSingleTapConfirmed.collect { event ->
+                    event.mapPoint?.let { point -> displayGeodesicPath(point) }
+                }
+            }.onFailure {
+                // if map load failed, show the error
+                showError("Error Loading Map")
+            }
+        }
+    }
+
+    /**
+     * Displays the destination location marker at the tapped location
+     * and draws a geodesic path curve using GeometryEngine.densifyGeodetic
+     * and computes the distance using GeometryEngine.lengthGeodetic
+     */
+    private fun displayGeodesicPath(point: Point) {
+        // project the tapped point into the same spatial reference as source point
+        val destinationPoint = GeometryEngine.project(point, SpatialReference.wgs84())
+
+        // check if the destination point is within the map bounds
+        // isEmpty returns true if out of bounds
+        if (!destinationPoint.isEmpty) {
+            // update the end location marker location on map
+            endLocationMarkerGraphic.geometry = destinationPoint
+            // create a polyline between source and destination points
+            val polyline = Polyline(listOf(startingPoint, destinationPoint))
+            // generate a geodesic curve using the polyline
+            GeometryEngine.densifyGeodetic(
+                polyline,
+                1.0,
+                unitsOfMeasurement,
+                GeodeticCurveType.Geodesic
+                // only compute the distance if the returned curved path geometry is not null
+            )?.let { pathGeometry ->
+                // update the path graphic
+                geodesicPathGraphic.geometry = pathGeometry
```

Comment:
If hypothetically this would return null then I guess we would also want to hide the geodesic path graphic by setting its geometry to null?